lib/runners/processor/golangci_lint.rb

```diff
@@ -0,0 +1,173 @@
+            parse_result(stdout).each { |v| result.add_issue(v) }
+          end
+        )
+      end
+
+      if status.exitstatus == 3
+        return Results::Failure.new(guid: guid, analyzer: analyzer, message: handle_respective_message(stderr))
+      end
+
+      if status.exitstatus == 5
+        return Results::Failure.new(guid: guid, analyzer: analyzer, message: "No go files to analyze")
+      end
+
+      Results::Failure.new(guid: guid, analyzer: analyzer, message: "Running error")
+    end
+
+    def handle_respective_message(stderr)
+      case stderr
+      when /must enable at least one/
+        "Must enable at least one linter"
+      when /can't be disabled and enabled at one moment/
+        "Can't be disabled and enabled at one moment"
+      when /can't combine options --disable-all and --disable/
+        "Can't combine options --disable-all and --disable"
+      when /only next presets exist/
+        "Only next presets exist: (bugs|complexity|format|performance|style|unused)"
+      when /no such linter/
+        "No such linter"
+      when /can't combine option --config and --no-config/
+        "Can't combine option --config and --no-config"
+      else
+        msg = stderr.match(/level=error msg=.+\[(.+)\]/)
+        msg && msg[1] ? msg[1] : "Running Error"
+      end
+    end
+
+    def config
+      @config or raise "Must be initialized!"
+    end
+
+    def analyzer_options
+      [].tap do |opts|
+        analysis_targets.each { |target| opts << target }
+        opts << "--out-format=json"
+        opts << "--issues-exit-code=0"
+        opts << "--tests=#{config[:tests]}" unless config[:tests].nil?
+        opts << "--config=#{path_to_config}" if path_to_config
+        Array(config[:disable]).each { |disable| opts << "--disable=#{disable}" }
+        Array(config[:enable]).each { |enable| opts << "--enable=#{enable}" }
+        Array(config[:presets]).each { |preset| opts << "--presets=#{preset}" }
+        Array(config[:'skip-dirs']).each { |dir| opts << "--skip-dirs=#{dir}" }
+        Array(config[:'skip-files']).each { |file| opts << "--skip-files=#{file}" }
+        opts << "--uniq-by-line=#{config[:'uniq-by-line']}" unless config[:'uniq-by-line'].nil?
+        opts << "--no-config=#{config[:'no-config']}" unless config[:'no-config'].nil?
+        unless config[:'skip-dirs-use-default'].nil?
+          opts << "--skip-dirs-use-default=#{config[:'skip-dirs-use-default']}"
+        end
+        opts << "--disable-all=#{config[:'disable-all']}" unless config[:'disable-all'].nil?
+      end
+    end
+
+    def path_to_config
+      return config[:config] if config[:config]
+
+      # @see https://github.com/golangci/golangci-lint#config-file
+      candidates = %w[.golangci.yml .golangci.toml .golangci.json].map { |filename| current_dir / filename }
+
+      found = candidates.find(&:exist?)
+      found ? found : "sider_golangci.yml".tap { |file| FileUtils.cp(Pathname(Dir.home) / file, current_dir) }
```

Comment:
[IMO] Sorry, it seems better to me to return `nil` (that is, omits `--config`) if a `.golangci.*` file is found. By doing so, we can use the configuration search mechanism of GolangCI-Lint itself.

```suggestion
      return if found
      "sider_golangci.yml".tap { |file| FileUtils.cp(Pathname(Dir.home) / file, current_dir) }
```

For example, if a user would have both `.golangci.yml` and `.golangci.json`, we could not determine which we should specify `--config=.golangci.yml` or `--config=.golangci.json`.